### ISSUE TITLE
Template variables not substituted on services section of configuration

### ISSUE THREAD

**Issue by thiagohmoreira** [mooltiverse/nyx#186]
I'm using Nyx v2.3.0 on command line.

My `.nyx.yaml` looks like:

```
preset: extended

...

git:
  remotes:
    origin:
      user: "PRIVATE-TOKEN"
      password: "{{#environmentVariable}}RELEASE_TOKEN{{/environmentVariable}}"

...

services:
  gitlab:
    type: GITLAB
    options:
      BASE_URI: "{{#environmentVariable}}CI_API_V4_URL{{/environmentVariable}}"
      AUTHENTICATION_TOKEN: "{{#environmentVariable}}CI_JOB_TOKEN{{/environmentVariable}}"
      REPOSITORY_NAME: "{{#environmentVariable}}CI_PROJECT_NAME{{/environmentVariable}}"
      REPOSITORY_OWNER: "{{#environmentVariable}}CI_PROJECT_NAMESPACE{{/environmentVariable}}"
```

And I can see this on the logs (ran it with `--trace` option):
```
...
time="2023-03-20T16:15:35Z" level=debug msg="publishing version '0.2.0.dev1' to 'gitlab'"
time="2023-03-20T16:15:35Z" level=trace msg="retrieving the services"
time="2023-03-20T16:15:35Z" level=debug msg="resolving the service configuration among available ones: 'map[gitlab:0xc00020d220]'"
time="2023-03-20T16:15:35Z" level=debug msg="instantiating service 'gitlab' of type 'GITLAB' with '4' options"
time="2023-03-20T16:15:35Z" level=trace msg="instantiating new GitLab service"
time="2023-03-20T16:15:35Z" level=trace msg="instantiating new GitLab client"
time="2023-03-20T16:15:35Z" level=debug msg="the new GitLab service will use the given authentication token"
time="2023-03-20T16:15:35Z" level=trace msg="the new GitLab service uses the custom URI '{{#environmentVariable}}CI_API_V4_URL{{/environmentVariable}}'"
time="2023-03-20T16:15:35Z" level=debug msg="publishing GitLab release '0.2.0.dev1'"
time="2023-03-20T16:15:35Z" level=debug msg="an error occurred while publishing GitLab release '0.2.0.dev1': Post \"%7B%7Bapi/v4/projects/%7B%7B%23environmentVariable%7D%7DCI_PROJECT_NAMESPACE%7B%7B/environmentVariable%7D%7D/%7B%7B%23environmentVariable%7D%7DCI_PROJECT_NAME%7B%7B/environmentVariable%7D%7D/releases#environmentVariable%7D%7DCI_API_V4_URL%7B%7B/environmentVariable%7D%7D/\": unsupported protocol scheme \"\""
could not publish GitLab release with tag '0.2.0.dev1': Post "%7B%7Bapi/v4/projects/%7B%7B%23environmentVariable%7D%7DCI_PROJECT_NAMESPACE%7B%7B/environmentVariable%7D%7D/%7B%7B%23environmentVariable%7D%7DCI_PROJECT_NAME%7B%7B/environmentVariable%7D%7D/releases#environmentVariable%7D%7DCI_API_V4_URL%7B%7B/environmentVariable%7D%7D/": unsupported protocol scheme ""
```

I didn't tried it with Gradle, but I see something similar [here](https://gitlab.com/kiscord/kitty/-/blob/master/.nyx.yaml) and seems to be ok. But on my case the variables are not substituted as expected. 

Please note that on the Git section it works fine (added that for reference). I'm also sure that all the referenced ENV variables are indeed defined and have proper values.

Did I miss something or I struck a mean little bug? ^^

**Comment by flelli** [comment]
Hi @thiagohmoreira ,

I'll look into this asap, actually looks like a bug. I'll keep you posted.

**Comment by flelli** [comment]
Hi @thiagohmoreira ,

I published the [2.3.1-hotfix186.1](https://github.com/mooltiverse/nyx/releases/tag/2.3.1-hotfix186.1) hotfix. Can you give it a try before I merge to main?

**Comment by thiagohmoreira** [comment]
Hello @flelli,

Works like a charm! Amazing response speed mate! Thanks a lot.

On another subject, I might want/need to implement another versioning schema, to support [PEP440 ](https://peps.python.org/pep-0440/), as it is not 100% compatible with SemVer (🤦‍♂️! ). I would like to get in touch with you about this, hear suggestions and get some implementation guidelines and what not. What you suggest as best way to do it? Create another issue with the proposal, and we discuss it there?

Once again, thank you very much!
